pkg/execution/state/driver_response.go

```diff
@@ -431,6 +431,34 @@ func (r *DriverResponse) HistoryVisibleStep() *GeneratorOpcode {
 	return op
 }
 
+// TraceVisibleStepExecution returns a single generator op if this response
+// should be visible in a trace, otherwise nil. If returning nil, the response
+// may still be considered to be a function response, in which case it likely
+// also needs to be tracked.
+func (r *DriverResponse) TraceVisibleStepExecution() *GeneratorOpcode {
+	if r.Generator == nil {
+		return nil
+	}
+
+	if len(r.Generator) != 1 {
```

Comment:
probably could just combine this with the one on top and make it an `||`

Reply:
Added some comments instead, as the checks are nice to have separate.